### PR TITLE
feat: custom line parser

### PR DESCRIPTION
**Pull request by climba03003**
Allow customize the line parser.

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by climba03003** [comment]
I may need some update to handle the `worker` problem about `function`.

**Comment by mcollina** [comment]
Maybe try rebasing?

**Comment by mcollina** [comment]
That option would not work when used it through a transport. However an user should use the module programmatically and create/link their own transport to pass in that function. Could you adjust docs/test?

File: README.md [hunk 1]
```diff
@@ -48,20 +48,29 @@ log.info({ $and: [{ a: 1 }, { b: 2 }] }, 'my query is')
...
 
     return obj // return value is expected to be a json that will pass and save inside mongodb
   }
+  return transport(opts)
+}
+
+// main.js
+const pino = require('pino')
+const transport = pino.transport({
+  target: 'mongo-transport.js',
```
Comment (mcollina) [inline README.md:71]:
```suggestion
  target: './mongo-transport.js',
```

File: README.md [hunk 2]
```diff
@@ -48,20 +48,29 @@ log.info({ $and: [{ a: 1 }, { b: 2 }] }, 'my query is')
...
 ```js
-const pino = require('pino')
-const transport = pino.transport({
-  uri: 'mongodb://localhost:27017/logs',
-  collection: 'log-collection',
-  parseLine(str) { // `str` is passed from `pino` and expected to be a string
+// mongo-transport.js
+const transport = require('pino-mongo')
+
+module.exports = async function(opts) {
```
Comment (mcollina) [inline README.md:57]:
```suggestion
module.exports = function(opts) {
```

File: test/fixture/wrap-transport.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const transport = require('../../lib/pino-transport')
```
Comment (mcollina) [inline test/fixture/wrap-transport.js:1]:
```suggestion
'use strict'

const transport = require('../../lib/pino-transport')
```

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**